### ISSUE TITLE
Balance showing as 0

### ISSUE THREAD

**Issue by JASHIKO** [electroneum/electroneum#19]
I have been using this wallet since day one. And have synced it fully today 2/12/2017 and my balance is showing as 0 after quite a fair bit of mining. Weeks and weeks worth, I'm certain the pools have paid out the figures as they are quite high. 

I have noticed this is a major problem as there are a lot of people having the issue on forums/youtube videos. 

Has all this mining been wasted?

**Comment by electroneumRepo** [comment]
From what I have seen, this is usually user error.

Please ensure you have the daemon fully synced and open whilst checking your wallet balance otherwise it will simply not work.

If you believe your figures are incorrect, re-import your wallet and let it rescan the blockchain. If it's still not right then delete the blockchain data directory (This may be under C:\ProgramData\Electroneum or yours users AppData folder). Then re-sync with the blockchain and try again.

Without you giving me your private keys, we cannot look into this any further. There are thousands of users that are mining without any issues and sending funds. There are many pools that are paying the miners, again without issues. If your using a specific pool, try switching to a different pool and see what happens there.

I'm sorry that this may not be the reply you was hoping for.

**Comment by TriJetScud** [comment]
I am seeing this issue occur with official Windows builds, even with a fully synchronized electroneum node that is running on Windows.

However, once I compiled electroneum on a Ubuntu 16.04 WSL userland, the wallet does appear to show the balances I've received from the mining pool I'm using.

**Comment by electroneumRepo** [comment]
@TriJetScud Have you tried:

1. `rescan_bc` on the windows cli?
2. Have you tried removing the blockchain and re-syncing?
3. Have you tried re-importing the wallet?

You said it worked after compiling on Ubuntu, Have you tried the Official Compiled binaries for linux?
Does the problem exist with those?

**Comment by bet0x** [comment]
The problem happens if you create your wallet with the tool on windows, i sent several ETN to my wallet and lost em. I made a paper one and it worked.

**Comment by Bonefroz** [comment]
@electroneumRepo 
i have the same issue on windows
1- doesn't work
2- re-syncing the enitre blockchain doesn't work
3- how can i re-import my wallet?

**Comment by JASHIKO** [comment]
Ok, so I spent a ton of time figuring this one out! And basically, there is a bug or something in the wallet or a compatibility issue with Windows...  (I'm by no means an expert though)  I did manage to get my view balance, however, after I closed the wallet and re-opened it the balance would show as 0 again... And I had to repeat a massive process..... The second time I decided to simply move the coin an exchange.  

For anyone else with the same issue check out the guide below. Let me know if it works for you.

https://www.downloadsource.net/how-to-fix-electroneum-etn-wallet-showing-0-balance/n/11022/

**Comment by WalterMagnum** [comment]
I am having the same issue on Windows as well.  Did I just lose 1000 ENT?  If the Windows wallet is this unreliable/buggy, I don't see mass adoption ever happening.

**Comment by oleersoy** [comment]
Same issue on Ubuntu:
https://github.com/electroneum/electroneum/issues/32

**Comment by TriJetScud** [comment]
Not experiencing this issue with my builds compiled from the master. Perhaps should we hard code in checkpoints in the daemon source code so less nodes don't get confused on which chain should be the majority chain?

**Comment by oleersoy** [comment]
The [MPOS](https://github.com/MPOS/php-mpos) pool shows payout transactions roughly every half hour and updates balances correspondlngly.

**Comment by sicruise** [comment]
Same issue on Ubuntu, made a transfer from one wallet to another. It shows the transfers on one wallet but the receiving wallet just doesn't get the coins.

I have double checked the ETN address.

**Comment by asteriskmonkey** [comment]
same issue, no one have a clue yet?

**Comment by johnferrie** [comment]
I am having the same issue on a mac.

`uptime 4d 13h 26m 11s`

`Balance: 0.00, unlocked balance: 0.00`

**Comment by sicruise** [comment]
Regenerating the wallet and syncing from scratch did the job for me. Any wallet I opened with electroneum-wallet-rpc just seemed to be corrupted.

**Comment by tahv0** [comment]
Having same problem:
```
[wallet etnkG2]: refresh
Starting refresh...
Refresh done, blocks received: 0                                
Balance: 0.00, unlocked balance: 0.00
```
Have I lost my money?

**Comment by johnferrie** [comment]
@sicruise is there a way to see if the wallet is corrupt? Did I just waste a week of mining with 0 results?

**Comment by bet0x** [comment]
@tahv0 it's seems they have no idea, i can import a paper wallet but can't export to recreate the one made with the cli?

**Comment by TriJetScud** [comment]
If you still have your electroneum wallet seeds, restore them from scratch on any other platform other than Windows and preferably a self-compiled build of the daemon and wallet cli and hopefully you should be able to see your balances there.

Even better, if you do know what transaction hashes that your wallet had for receiving the coins, you should be able to verify it by using a block explorer over at https://etnexplorer.info or https://blockexplorer.electroneum.com.

**Comment by tahv0** [comment]
@TriJetScud I'm on Ubuntu and I've compiled this repository myself (master branch). So this is not WIndows related issue.
I've wallet seeds and haven't tried yet tried to reload blockchain with electroneumd and recreate wallet with electroneum-wallet-cli. Will try later.

**Comment by sicruise** [comment]
Workaround -

1. Load up wallet
2. Use command 'seed'
3. Use output to regenerate a new wallet with command 'electroneum-wallet-cli --restore-deterministic-wallet'
4. Allow it to sync

You should now see your correct balance and be able to perform transfers :-)

**Comment by carlosearaujo** [comment]
I had the same problem and solved it by luck. I left the daemon and wallet open all night and the balance was right :)

**Comment by 1RedOne** [comment]
Electroneum really should have an online blockchain master running, so we can do simple requests to validate a balance online.

**Comment by ScotSJI** [comment]
@1RedOne  Do you even get the purpose of a decentralized currency? 

The problem is you're not letting your node fully sync before starting the CLI wallet.  You need to let it sync all blocks before initiating any wallet actions.    It takes forever, and is much quicker on Linux (which I'm sadly not on).

**Comment by 1RedOne** [comment]
It turns out that the wallet was corrupted and I needed to do a
deterministic restore using my seed words.  I'm not sure why that is
necessary, but it worked.

**Comment by Concorde89** [comment]
All works using electronum-wallet-cli if you use electroneum-wallet-cli --restore-deterministic-wallet only once. Use the command, input seed, start resyncing from a high block number (not neccesary to start from the beginning). Once this is done always use "exit" command when exiting wallet. You wont lose balance if you use it this way.

**Comment by CryptoGmoney** [comment]
I mined some ETN using the Electroneum Miner and the eupool the miner paid out 53ETN but when you click on the TX transaction hash it says it cant find TX. I have also tried to transfer the balance from my offline paper wallet to the webwallet but it says the wallet is empty

![mining etn proof of 53etn paid but tx cant find](https://user-images.githubusercontent.com/35033741/34489653-77c522f8-efdd-11e7-9626-b91789b4f9de.jpg)

**Comment by demirelo** [comment]
@CryptoGmoney I have a similar issue. Did you get any updates on this?

**Comment by CryptoGmoney** [comment]
No not yet. Have you?

On 7 Jan 2018 11:43 pm, demirelo <notifications@github.com> wrote:

@CryptoGmoney<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcryptogmoney&data=02%7C01%7C%7Cc9b377058e234a0f3b3c08d556201230%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636509618094734527&sdata=N8y7M2P8nT69qeK5jh4Pxq1Cn4rY6886zq4LvWtxSPg%3D&reserved=0> I have a similar issue. Did you get any updates on this?

**Comment by demirelo** [comment]
@CryptoGmoney Nope. This is probably affecting many (ex-) eupool users. It might be worth opening a new issue.

**Comment by CryptoGmoney** [comment]
Where is the best place to raise the subject? Email? Facebook? Twitter? Discord - Which hashtag?

On 7 Jan 2018 11:58 pm, demirelo <notifications@github.com> wrote:

@CryptoGmoney<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcryptogmoney&data=02%7C01%7C%7C5b304c5c2f5f4dbedcdd08d5562229bf%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636509627093276661&sdata=Ot8yklRPMcq0em9V2nETV2uygd9atMTdkA6FTqW3cWU%3D&reserved=0> Nope. This is probably affecting many (ex-) eupool users. It might be worth opening a new issue.

**Comment by quickdraw22** [comment]
I have been working on this issue on Mac. The command is not recognized. Please help

**Comment by avi84** [comment]
Hi i made an electroneum wallet but when I check the view key it says this:

The account does not exist
Remember: to add your account to the database you should find at least one share

Can any body help?

**Comment by CryptoGmoney** [comment]
Have you got an account? Signed up and Registered on the website? Maybe you have to sign up before you can set up a wallet? IMO you shouldn't have to sign up to get a wallet as it's a privacy coin but if you want customer service which is a bit lacking at the moment due to scalability then you need to sign up. It's a new problem I've never come across so far.

On 9 Jan 2018 7:58 am, avi84 <notifications@github.com> wrote:

Hi i made an electroneum wallet but when I check the view key it says this:

The account does not exist
Remember: to add your account to the database you should find at least one share

Can any body help?

**Comment by brisbanewebdeveloper** [comment]
I am using macOS and https://github.com/electroneum/electroneum/issues/19#issuecomment-351996789 worked (It took time to do re-sync, though). Thank you!

**Comment by MarQuisKnox** [comment]
I can confirm that this is still an issue that needs resolution. 

I compiled from the master 2 days ago on OS X Sierra, 10.12.6 & had a balance of 0 after receiving 30+ ETN. Following [the steps](https://github.com/electroneum/electroneum/issues/19#issuecomment-354298976) mentioned above by @Concorde89 worked for me.

This is unacceptable & should be resolved immediately. There are even "crap coins" that do not have this issue. Just saying...

**Comment by MarQuisKnox** [comment]
~~I archived the URL on the Wayback Machine for redundancy:  http://web.archive.org/web/20180131191054/https://www.downloadsource.net/how-to-fix-electroneum-etn-wallet-showing-0-balance/n/11022/~~ (not the best method, because the balance does not remain in your new wallet after exit)

**Comment by frankcoderL** [comment]
@electroneumRepo and all other issue reporters

Had the same issue on Ubuntu 16.04 LTS, and i have never experienced this problem on other  Cryptonight wallets/nodes.
**Please fix this bug, it's not a user error (at least not in my case). This bug is unacceptable.
Thanks!**


The issues that i have experienced:
Created a electroneum wallet file after compiling the master code. The balance stays always 0.0 , but there are coins at the wallet adress. `rescan_bc` is executed in 3 seconds what seems to be very short for scanning the 10 gigabyte blockchain, the balance stays at 0.0 after `rescan_bc`

After using `electroneum-wallet-cli --restore-deterministic-wallet`  to create a new wallet file i can see the coins (about one hour needed for scanning the 10 gigabyte blockchain).

Now for the same electroneum adress i have two wallet files, the first wallet file says always 0.0 balance, and the new wallet file that shows the correct balance.

**Comment by PetrovZoltan** [comment]
Same problem here. I tried everything, but the balance is zero. Any solution?

**Comment by JASHIKO** [comment]
https://www.downloadsource.net/how-to-fix-electroneum-etn-wallet-showing-0-balance/n/11022/

^
I posted this up the top, it worked well for me to get my online wallet balance then move it to an exchange etc. However, I still have an issue with a large portion of early ETN mining that hasn't been paid (official EU pool), tickets pending since early December) I get a reply every so often about it. The ETN team seem flat out though. 

I hope that helps Petrov.

**Comment by mkrufky** [comment]
For the record, I had a similar problem when I had tried mining Stellite a few months ago (using Stellite's v0.0.8 CLI ~April 20). The payments simply never showed up in the wallet. Without a solution, I had just given up and moved on to other coins. After `--restore-deterministic-wallet` solved this issue today with my Electroneum wallet, I decided to give it a try with the old Stellite wallet as well. Lo and behold, my missing Stellite all shows up properly in the newly restored wallet.

This bug must have come from the original Monero wallet that both Stellite and Electroneum wallets have forked from.
